sdk/data/azcosmos/cosmos_global_endpoint_manager_test.go

```diff
@@ -0,0 +1,407 @@
+
+	preferredRegions := []string{"West US", "Central US"}
+
+	serverEndpoint, err := url.Parse(srv.URL())
+	assert.NoError(t, err)
+
+	mockLc := newLocationCache(preferredRegions, *serverEndpoint)
+	mockLc.enableMultipleWriteLocations = true
+	mockLc.useMultipleWriteLocations = true
+
+	mockGem := globalEndpointManager{
+		client:              client,
+		preferredLocations:  preferredRegions,
+		locationCache:       mockLc,
+		refreshTimeInterval: 5 * time.Minute,
+	}
+
+	gem, err := newGlobalEndpointManager(client, preferredRegions, 5*time.Minute)
+	assert.NoError(t, err)
+
+	// Multiple locations should be false for default GEM
+	canUseMultipleWriteLocs := gem.CanUseMultipleWriteLocations()
+	assert.False(t, canUseMultipleWriteLocs)
+
+	// Mock GEM with multiple write locations available should show true
+	canUseMultipleWriteLocs = mockGem.CanUseMultipleWriteLocations()
+	assert.True(t, canUseMultipleWriteLocs)
+}
+
+func TestGlobalEndpointManagerUpdate(t *testing.T) { //This test should be testing for the lock on the update/ refresh mechanism
+	// Create a mock client
+	headerPolicy := &headerPolicies{}
+	srv, close := mock.NewTLSServer()
+	defer close()
+	srv.SetResponse(mock.WithStatusCode(http.StatusOK))
+
+	westRegion := accountRegion{
+		Name:     "West US",
+		Endpoint: srv.URL(),
+	}
+
+	properties := accountProperties{
+		ReadRegions:                  []accountRegion{westRegion},
+		WriteRegions:                 []accountRegion{westRegion},
+		EnableMultipleWriteLocations: false,
+	}
+
+	jsonString, err := json.Marshal(properties)
+	if err != nil {
+		t.Fatal(err)
+	}
+	srv.SetResponse(mock.WithBody(jsonString))
+
+	verifier := headerPoliciesVerify{}
+	pl := azruntime.NewPipeline("azcosmostest", "v1.0.0", azruntime.PipelineOptions{PerCall: []policy.Policy{headerPolicy, &verifier}}, &policy.ClientOptions{Transport: srv})
+	req, err := azruntime.NewRequest(context.Background(), http.MethodGet, srv.URL())
+	assert.NoError(t, err)
+
+	req.SetOperationValue(pipelineRequestOptions{
+		isWriteOperation: true,
+	})
+
+	client := &Client{endpoint: srv.URL(), pipeline: pl}
+
+	gem, err := newGlobalEndpointManager(client, []string{}, 5*time.Minute)
+	assert.NoError(t, err)
+
+	// Update the location cache and client's default endpoint
+	err = gem.Update()
+	assert.NoError(t, err)
+
+	// Get the write endpoints after the update
+	writeEndpoints, err := gem.GetWriteEndpoints()
+	assert.NoError(t, err)
+
+	serverEndpoint, err := url.Parse(srv.URL())
+	assert.NoError(t, err)
+
+	// Assert the expected write endpoints after the update
+	expectedWriteEndpoints := []url.URL{
+		*serverEndpoint,
+	}
+	assert.Equal(t, expectedWriteEndpoints, writeEndpoints)
+
+	// Get the read endpoints after the update
+	readEndpoints, err := gem.GetReadEndpoints()
+	assert.NoError(t, err)
+
+	// Assert the expected read endpoints
+	expectedReadEndpoints := []url.URL{
+		*serverEndpoint,
+	}
+	assert.Equal(t, expectedReadEndpoints, readEndpoints)
+}
+
+// Emulator Test
```

Comment:
This comment should be removed?

Reply:
was just using it to mark the difference between the tests but I'll remove it

Reply:
Emulator tests should be in a separate file that starts with `emulator_......`